lms/djangoapps/courseware/masquerade.py

```diff
@@ -229,6 +236,94 @@ def get_masquerade_role(user, course_key):
     return course_masquerade.role if course_masquerade else None
 
 
+def _get_masquerade_gating_group_id(user, course_key):
```

Comment:
Soo, after waaay too much spelunking, there turn out to be a lot of misleading helpers that either:
- don't quite do what we need or what they say
- require spinning up some "machinery" we aren't already pulling into the API (examples of querying `user_partitions` coming from a `modulestore()` backed course lookup), that helpers expect to be passed along.

Since masquerade settings are stored as integers and we have access to constants for reference, we benefit from a lighter-weight lookup.